### PR TITLE
[ui] Elapsed time indicators in log

### PR DESCRIPTION
**Pull request by mugulmd**
## Description

The goal of this PR is to enrich the node log viewer so that finding the steps that require long calculations is easier for users. To achieve that, we retrieve the time value of each log-line (when it exists) and we display the elapsed time between consecutive time values.

We add 2 visual indicators: 
- a colored marker next to the line number
- a tooltip when hovering over the line number cell displaying the elapsed time.

### CODE REVIEW

**Comment by mugulmd** [comment]
For now there are still 2 main issues to work on that I noted: 
- the color scale for the markers is arbitrary and definitely requires some fine-tuning
- when changing the chunk while displaying the log, the following QML warning appear: `TextFileViewer.qml:274: TypeError: Value is undefined and could not be converted to an object`

**Comment by fabiencastan** [comment]
The colorbar should have the line height:
image

**Comment by fabiencastan** [comment]
Time in tooltip should be formatted h:m:s.
The zone to display the Tooltip should include the colorbar.

**Comment by fabiencastan** [comment]
Should interpolate colors.

**Comment by fabiencastan** [comment]
The colorbar should be grey if the timing is 0 (and not the same than the background), so we can always see a continuous vertical line.

File: meshroom/ui/qml/Utils/Colors.qml [hunk 1]
```diff
@@ -61,17 +61,21 @@ QtObject {
         return "magenta"
     }
 
-    function durationColorScale(t) {
```
Comment (fabiencastan) [inline meshroom/ui/qml/Utils/Colors.qml]: You should interpolate between: grey > green > yellow > red.

File: meshroom/ui/qml/Controls/TextFileViewer.qml [hunk 2]
```diff
@@ -203,7 +203,7 @@ Item {
                     Rectangle {
                         width: 4
                         Layout.fillHeight: true
-                        color: duration > 0 ? Colors.durationColorScale(duration) : "transparent"
+                        color: Colors.interpolate(Colors.grey, Colors.red, duration/180)
```
Comment (fabiencastan) [inline meshroom/ui/qml/Controls/TextFileViewer.qml]: Need to use Colors.durationColorScale to get something visible. Just add interpolation in it.

**Review by fabiencastan — APPROVED** [review]

**Pull request merged**